tests/prague/eip2935_historical_block_hashes_from_state/test_block_hashes.py

```diff
@@ -364,3 +375,63 @@ def test_invalid_history_contract_calls(
         post=post,
         reverts=reverts,
     )
+
+
+@pytest.mark.parametrize(
+    "args_size,reverts",
+    [
+        pytest.param(0, True, id="zero_size"),
+        pytest.param(33, True, id="too_large"),
+        pytest.param(31, True, id="too_small"),
+    ],
+)
+@pytest.mark.valid_from("Prague")
+def test_invalid_history_contract_calls_input_size(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    reverts: bool,
+    args_size: int,
+):
+    """Test calling the history contract with invalid input sizes."""
+    storage = Storage()
+
+    return_code_slot = storage.store_next(not reverts)
```

Comment:
you can add comment for debugging if there would be storage mismatch

storage.store_next(not reverts, "system call result")

Reply:
Thanks, will add this.